csharp-design-patterns/UnitOfWork/UnitOfWork/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,24 @@
+using UnitOfWork.DataAccess;
+using UnitOfWork.Entities;
+
+namespace UnitOfWork.Repositories;
+
+public class OrderRepository : IOrderRepository
+{
+    private readonly IDatabase _database;
+
+    public OrderRepository(IDatabase database)
+    {
+        _database = database;
+    }
+
+    public void Add(Order order)
+    {
+        _database.AddOrder(order);
+    }
+}
+
+public interface IOrderRepository
```

Comment:
I would move this `IOrderRepository` into a separate file. In general we should avoid having multiple classes/interfaces/records in one file.

Reply:
moved